Fix multiple localisation issues

tukusejssirs opened the pull request:
- create:
   - a new key for St Henry (Emperor), celebrated on 13 July in `general.ts`, as currently his key is merged with St Henry, bishop and martyr, who is celebrated on 19 January in Scandinavia only;
- add:
   - titles to `saintApollinaris` value in `en.ts` (he was a bishop and a martyr);
   - `saintsJohnDeBrebeufIsaacJoguesPriestsAndCompanionsMartyrsSecondaryPatronsOfCanada` to `en.ts` and `sk.ts`;
   - `sundayOfTheWordOfGod`, `saintRoseOfLimaSecondaryPatronessOfThePhilippines`, `ourLadyOfGuadalupeCelestialPatronessOfThePhilippines`, `saintMariaGuadalupeGarciaZavalaVirgin` and `ourLadyOfRefuge` keys to `en.ts` and `sk.ts`;
- remove:
   - `saintJohnOfTheCrossDoctorOfTheChurch` in favour of `saintJohnOfTheCrossPriestAndDoctor`;
   - `saintsAugustineZhaoRongPriestAndCompanionsMartyrs` in favour of `saintAugustineZhaoRongPriestAndCompanionsMartyrs`;
   - `saintDenisAndCompanionsMartyrs` celebration definition from `poland.ts` and `portugal.ts` as it looks like there (in `poland.ts`) is typo in the date (day of month and month are switched) and (in both calendars) it is not listed in the [Wiki](https://en.wikipedia.org/wiki/General_Roman_Calendar#Poland);
   - `en-CA.ts` as it contained only `saintsJeanDeBrebeufAndIsaacJoguesPriestsAndCompanionsMartyrs` and it was moved to `en.ts` (until we deal with #201);
   - `saintBridgetOfSwedenReligious` from `venezuela.ts` and `uruguay.ts`;
- rename:
   - `ourLadyOfPerpetualHelpOrBlessedLeonidFeodorovPriestAndMartyr` key to `ourLadyOfPerpetualHelp`  in `pt-BR.ts`, as it is the key value and the original key it not used in any calendar;
   - `saintsJeanDeBrebeufAndIsaacJoguesPriestsAndCompanionsMartyrs` to `saintsJohnDeBrebeufIsaacJoguesPriestsAndCompanionsMartyrs`;
   - the value of the `maryMotherOfTheChurch` key from `Mary, Mother of The Church` to `Mary, Mother of the Church` in `en.ts`;
   - `SaintPirminAbbotAndBishop` to `saintPirminAbbotAndBishop`;
   - the value of the `saintMaryMagdaleneDePazziVirgin` key from `Saint Mary Magdalene de Pazzi, Virgin` to `Saint Mary Magdalene de’ Pazzi, Virgin` in `en.ts`;
   - `saintJohnPaulIiPope` and `popeSaintJohnXXIII` from `italy.ts` (as they are already defined in `general.ts`);
- replace:
   - missed `saintColumbanAbbotAndMissionary` with `saintColumbaAbbotAndMissionary` in l10n files;
   - `saintsJeanDeBrebeufIsaacJoguesPriestsAndCompanionsMartyrs` with `saintsJohnDeBrebeufIsaacJoguesPriestsAndCompanionsMartyrsSecondaryPatronsOfCanada` in `canada.ts`;
   - `saintElizabethOfHungary` with `saintElizabethOfHungaryReligious` in some calendars;
   - `saintHedwigOfSilesia` with `saintHedwigReligious` in `poland.ts`;
   - `saintMargaretMaryAlacoqueVirgin` with `saintMargaretMaryAlacoqueVirgin`;
   - `saintBridgetOfSwedenReligious` with `saintBrigidVirgin` in `ireland.ts`;
   - `ourLadyOfMountCarmel` with `ourLadyOfMountCarmelMotherAndQueenOfChile` in `chile.ts`;
   - `saintPaulMikiAndCompanionsMartyrs` with `saintsFelipeDeJesusPaulMikiAndCompanionsMartyrs` in `bolivia.ts`;
   - the value of the `saintGregoryViiPope` key from `Saint Bede the Venerable, Priest and Doctor` to `Saint Gregory VII, Pope` in `en.ts`;
- separate:
   - `saintsFabianPopeAndSebastianMartyrs` and `saintJeromeEmilianiSaintJosephineBakhitaVirgin`;
   - `saintJeromeEmilianiSaintJosephineBakhitaVirgin` celebration in `general.ts`;
- localise keys that I have missed in `sk.ts`;
- localise colours in `sk.js`;
- update `philippines.ts` and `mexico.ts` calendars (remove duplicate definitions and add missing celebration);
- change:
   - `saintBridgetOfSwedenReligious` rank definition in `general.ts` from feast to optional memorial;
   - value of `ourLadyOfThePillar` from `Our Lady of The Pillar` to `Our Lady of the Pillar`;
   - value of `ourLadyOfTheValley` from `Our Lady of The Valley` to `Our Lady of the Valley`;
   - value of `ourLadyOfTheGateOfDawn` from `Our Lady of The Gate of Dawn` to `Our Lady of the Gate of Dawn`;
   - the key and value of `popeSaintJohnPaulII` to `saintJohnPaulIiPope` and `Saint John Paul II, Pope`;
   - the key and value of `popeSaintJohnXXIII` to `saintJohnXxiiiPope` and `Saint John XXIII, Pope`;
- fix issues with St Columba and St Columban.

tukusejssirs committed:
Fix multiple localisation issues

- create a new key for St Henry (Emperor), celebrated on 13 July in `general.ts`, as currently his key is merged with St Henry, bishop and martyr, who is celebrated on 19 January in Scandinavia only;
- add titles to `saintApollinaris` value in `en.ts` (he was a bishop and a martyr);
- remove `saintJohnOfTheCrossDoctorOfTheChurch` in favour of `saintJohnOfTheCrossPriestAndDoctor`;
- remove `saintDenisAndCompanionsMartyrs` celebration definition from `poland.ts` and `portugal.ts` as it looks like there (in `poland.ts`) is typo in the date (day of month and month are switched) and (in both calendars) it is not listed in the [Wiki](https://en.wikipedia.org/wiki/General_Roman_Calendar#Poland);
- `sk.ts`: localised keys that I have missed;
- fix issues with St Columba and St Columban;
- `en.ts`: rename the value of the `maryMotherOfTheChurch` key from `Mary, Mother of The Church` to `Mary, Mother of the Church`;
- `en.ts`: rename the value of the `saintMaryMagdaleneDePazziVirgin` key from `Saint Mary Magdalene de Pazzi, Virgin` to `Saint Mary Magdalene de’ Pazzi, Virgin`;
- `en.ts`: change the value of the `saintGregoryViiPope` key from `Saint Bede the Venerable, Priest and Doctor` to `Saint Gregory VII, Pope`;
- separate `saintsFabianPopeAndSebastianMartyrs` and `saintJeromeEmilianiSaintJosephineBakhitaVirgin`.

tukusejssirs commented:
@CodingAleCR, I’d like to notify you that when this PR is merged, some of l10n keys change; see the diff for `en.ts` file.

CodingAleCR commented:
Will take them into account when I modify my PR. I might close the js one and just make the ts one

tukusejssirs commented:
> I might close the js one and just make the ts one with a new PR.

You can do that, but if you have rebased your branch , in which you originally worked, then it would be possible just to change the `master` branch to the TS branch. On the other hand, I have I’ve just noticed that you worked directly in the `master` branch of your fork. It might be better to always create a new branch when you change a code and when that branch is merged upstream, you would delete that branch. That way you could always your fork (the branches that you have in common with the upstream) in sync with upstream.

tukusejssirs committed:
Fix multiple localisation issues 2

- sort the keys in the `sanctoral` section of the l10n files;
- remove `saintsAugustineZhaoRongPriestAndCompanionsMartyrs` in favour of `saintAugustineZhaoRongPriestAndCompanionsMartyrs`;
- rename `SaintPirminAbbotAndBishop` to `saintPirminAbbotAndBishop`;
- localise colours in `sk.js`.

tukusejssirs committed:
Fix localisation issues 3

- rename `ourLadyOfPerpetualHelpOrBlessedLeonidFeodorovPriestAndMartyr` key to `ourLadyOfPerpetualHelp`  in `pt-BR.ts`, as it is the key value and the original key it not used in any calendar;
- rename `saintsJeanDeBrebeufAndIsaacJoguesPriestsAndCompanionsMartyrs` to `saintsJohnDeBrebeufIsaacJoguesPriestsAndCompanionsMartyrs`;
- replace missed `saintColumbanAbbotAndMissionary` with `saintColumbaAbbotAndMissionary` in l10n files;
- replace `saintsJeanDeBrebeufIsaacJoguesPriestsAndCompanionsMartyrs` with `saintsJohnDeBrebeufIsaacJoguesPriestsAndCompanionsMartyrsSecondaryPatronsOfCanada` in `canada.ts`;
- replace `saintElizabethOfHungary` with `saintElizabethOfHungaryReligious` in some calendars;
- replace `saintHedwigOfSilesia` with `saintHedwigReligious` in `poland.ts`;
- replace `saintMargaretMaryAlacoqueVirgin` with `saintMargaretMaryAlacoqueVirgin`;
- replace `saintBridgetOfSwedenReligious` with `saintBrigidVirgin` in `ireland.ts`;
- replace `ourLadyOfMountCarmel` with `ourLadyOfMountCarmelMotherAndQueenOfChile` in `chile.ts`;
- replace `saintPaulMikiAndCompanionsMartyrs` with `saintsFelipeDeJesusPaulMikiAndCompanionsMartyrs` in `bolivia.ts`;
- add `saintsJohnDeBrebeufIsaacJoguesPriestsAndCompanionsMartyrsSecondaryPatronsOfCanada` to `en.ts` and `sk.ts`;
- add `sundayOfTheWordOfGod`, `saintRoseOfLimaSecondaryPatronessOfThePhilippines`, `ourLadyOfGuadalupeCelestialPatronessOfThePhilippines`, `saintMariaGuadalupeGarciaZavalaVirgin` and `ourLadyOfRefuge` keys to `en.ts` and `sk.ts`;
- remove `en-CA.ts` as it contained only `saintsJeanDeBrebeufAndIsaacJoguesPriestsAndCompanionsMartyrs` and it was moved to `en.ts` (until we deal with #201);
- remove `saintBridgetOfSwedenReligious` from `venezuela.ts` and `uruguay.ts`;
- update `philippines.ts` and `mexico.ts` calendars (remove duplicate definitions and add missing celebration);
- separate `saintJeromeEmilianiSaintJosephineBakhitaVirgin` celebration in `general.ts`;
- change `saintBridgetOfSwedenReligious` rank definition in `general.ts` from feast to optional memorial.

tukusejssirs commented:
There are still many, many issues (be it l10n issues or calendar issues), but the debugging is quite laborious without proper l10n tool (and without separating the celebration metadata from the current l10n keys).

tukusejssirs committed:
Fix localisation issues 4

- change value of `ourLadyOfThePillar` from `Our Lady of The Pillar` to `Our Lady of the Pillar`;
- change value of `ourLadyOfTheValley` from `Our Lady of The Valley` to `Our Lady of the Valley`;
- change value of `ourLadyOfTheGateOfDawn` from `Our Lady of The Gate of Dawn` to `Our Lady of the Gate of Dawn`;
- change the key and value of `popeSaintJohnPaulII` to `saintJohnPaulIiPope` and `Saint John Paul II, Pope`;
- change the key and value of `popeSaintJohnXXIII` to `saintJohnXxiiiPope` and `Saint John XXIII, Pope`;
- remove `saintJohnPaulIiPope` and `popeSaintJohnXXIII` from `italy.ts` (as they are already defined in `general.ts`).

emagnier commented:
Hello @tukusejssirs, is it something ready to be merged (after review) or do you still have to work on it?

tukusejssirs commented:
@emagnier, yes, this PR is mergable. BTW, I always use `WIP` prefix when it is not yet complete.

If you asked this because of multiple commits, then my answer is: romcal still has many issues because it is not easily debuggable from the l10n point of view, as well as from the calendar definition point of view. For this very reason I impatiently wait for and look forward to the separation of the celebration metadata (#154) and adding a the localisation tool (#43).

emagnier commented:
Ok good, I will do a quick review as soon as I can 👍

emagnier commented on src/calendars/general.ts:
```diff
@@ -845,7 +860,7 @@ const dates = async (config: Config): Promise<Array<RomcalDateItemInput>> => {
       key: 'saintBridgetOfSwedenReligious',
-      type: TypesEnum.FEAST,
+      type: TypesEnum.OPT_MEMORIAL,
       date: dayjs.utc(`${year}-7-23`),
```
Are you sure of that? Would not be just for Europe?

tukusejssirs replied:
According the the [WIKI](https://en.wikipedia.org/wiki/General_Roman_Calendar#July), in `general.ts`, it is a mere optional memorials. In Europe, the celebration’s rank is raised to feast, as St Bridget of Sweden is patron of Europe.

Is your question related to something else?

emagnier replied:
Yes, sorry I misread your correction and saw the opposite.

tukusejssirs committed:
Fix prettier syntax errors

emagnier commented on src/calendars/ireland.ts:
```diff
@@ -38,7 +38,7 @@ const dates = async (config: Config): Promise<Array<RomcalDateItemInput>> => {
     },
     {
-      key: 'saintBridgetOfSwedenReligious',
+      key: 'saintBrigidVirgin',
```
Why are you reducing its name, like a few other. She is well know as Bridget of Sweden, so it's still useful to keep this information in his key.

tukusejssirs replied:
[St Bridget of Sweden](https://en.wikipedia.org/wiki/Bridget_of_Sweden) and [St Brigid of Kildare](https://en.wikipedia.org/wiki/Brigid_of_Kildare) are two different saints. And they are celebrated on different dates.

emagnier replied:
Understood, this isn't the same person.

emagnier commented on src/locales/index.ts:
```diff
@@ -11,7 +10,6 @@ import sk from './sk';
  * in romcal.
  */
 const locales = {
-  'en-CA': enCA,
```
You are removing the only one localisation file for a specific country/region.
In the top of this file, we might add a basic example (in a comment block) if someday we have to reintroduce one again, especially about the naming and the common conventions for this type of data (xx-XX and xxXX).

tukusejssirs replied:
Yeah, I am removing it as currently, it is not useful at all. All regional locales should contain only differences (in comparison with the ‘base’ locale) from the l10n/linguistic point of view; it should not create new keys.

> In the top of this file, we might add a basic example (in a comment block) if someday we have to reintroduce one again, especially about the naming and the common conventions for this type of data (xx-XX and xxXX).

Well, I see want you mean, however, why would be add a placeholder file like this into the romcal codebase? I think that better place for this is the documentation.

emagnier replied:
It's a very specific things and it's often useful to have this type of instructions directly in the codebase. So we don't have to search how to do it when we have to, especially when it's a code specific instruction. And also because we generally don't read documentations, until we are block since a moment :)

So I think it's a good thing to add just a very basic example here, and it will improve future contributions experience.

tukusejssirs replied:
From my point of view, it is not the best place. I think that if we once have a localisation tool _and_ a website/web-app that would make it possible to create a new locale, that note would be unnecessary.

If you, however, really need that example in the codebase, what about `src/locales/index.ts`? We could prepend the example their. It might be even sufficient to note that regional locales (like `en-CA.ts`) should have `xx-XX.ts` format.

> And also because we generally don't read documentations, until we are block since a moment :)

This is not the case with me. I always read the docs, man pages and any help I could get. :smiley: I really miss an ultimate documentation (be it user and dev documentation) of romcal.

> So I think it's a good thing to add just a very basic example here, and it will improve future contributions experience.

If you really insist, send me a patch or a text example.

emagnier replied:
I was speaking about adding an example just above in `src/locales/index.ts`. But after looking at the entire file, I just saw that we still have the 'pt-BR: ptBR` (I forgot it had been added recently). So, it's all right 👍

emagnier reviewed (changes requested):
I just have a 3 small comments that might need attention, other than that it's a LGTM 👍

emagnier commented:
Merged 🎉. Good works, thanks a lot @tukusejssirs !
